### PR TITLE
feat: add utils to extract v2/v3 template data

### PR DESCRIPTION
**Pull request by ChenRen93**
Update:  util to return unwrapped document data

### CODE REVIEW

File: src/shared/utils/utils.ts [hunk 1]
```diff
@@ -121,6 +122,28 @@ export const getTemplateURL = (document: any): string | undefined => {
...
+  document: T
+):
+  | {
+      [key: string]: any;
+    }[]
+  | {
+      [key: string]: any;
+    } => {
+  console.log(isWrappedV2Document(document));
+  console.log(isRawV2Document(document));
```
Comment (simboonlong) [inline src/shared/utils/utils.ts]: remove

File: src/shared/utils/utils.ts [hunk 2]
```diff
@@ -121,6 +122,26 @@ export const getTemplateURL = (document: any): string | undefined => {
   }
 };
 
+export const getTemplateData = <T extends WrappedDocument<OpenAttestationDocument> | OpenAttestationDocument>(
+  document: T
```
Comment (simboonlong) [inline src/shared/utils/utils.ts:126]: add tests for v2/v3 scenarios?

File: src/shared/utils/utils.ts [hunk 3]
```diff
@@ -121,6 +122,26 @@ export const getTemplateURL = (document: any): string | undefined => {
...
+):
+  | {
+      [key: string]: any;
+    }[]
+  | {
+      [key: string]: any;
+    } => {
+  if (isWrappedV3Document(document) || isRawV3Document(document)) {
+    return document.credentialSubject;
```
Comment (Nebulis) [inline src/shared/utils/utils.ts:135]: this is wrong =)

File: src/shared/utils/utils.ts [hunk 4]
```diff
@@ -121,6 +122,26 @@ export const getTemplateURL = (document: any): string | undefined => {
   }
 };
 
+export const getTemplateData = <T extends WrappedDocument<OpenAttestationDocument> | OpenAttestationDocument>(
```
Comment (Nebulis) [inline src/shared/utils/utils.ts:125]: wrong name ? it looks like the function returns the document data

File: src/shared/utils/utils.ts (hunk 3 above)
Comment (ChenRen93) [reply src/shared/utils/utils.ts:135]: why is this wrong? isn't wrapped and raw v3 document have the same schema and doesn't require unsalt data?

File: src/shared/utils/utils.ts [hunk 5]
```diff
@@ -122,23 +122,15 @@ export const getTemplateURL = (document: any): string | undefined => {
...
-    }[]
-  | {
-      [key: string]: any;
-    } => {
-  if (isWrappedV3Document(document) || isRawV3Document(document)) {
-    return document.credentialSubject;
+export const getDocumentData = (document: WrappedDocument<OpenAttestationDocument>): OpenAttestationDocument => {
+  if (isWrappedV3Document(document)) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { proof, ...rest } = document;
```
Comment (Nebulis) [inline src/shared/utils/utils.ts:128]: try to rename the variable with _: `proof: _whatever`, should fix eslint I think

Comment (ChenRen93) [reply src/shared/utils/utils.ts:128]: it doesn't fix the eslint, i will use lodash omit

File: src/shared/utils/__tests__/utils.test.ts [hunk 6]
```diff
@@ -363,4 +363,50 @@ describe("Util Functions", () => {
...
+    });
+
+    it("should return document data for v3 wrapped document", () => {
+      const v3WrappedDocument = v3WrappedVerifiableDocument as WrappedDocument<OpenAttestationDocument>;
+      expect(utils.getDocumentData(v3WrappedDocument)).not.toHaveProperty("proof");
+    });
+
+    test("should return error when document is not OpenAttestation document", async () => {
+      const document: any = {
+        RandomData: {
```
Comment (Nebulis) [inline src/shared/utils/__tests__/utils.test.ts:401]: Camel case

File: src/shared/utils/__tests__/utils.test.ts [hunk 7]
```diff
@@ -363,4 +363,50 @@ describe("Util Functions", () => {
...
+          "recipient": Object {
+            "name": "John Doe",
+          },
+        }
+      `);
+    });
+
+    it("should return document data for v3 wrapped document", () => {
+      const v3WrappedDocument = v3WrappedVerifiableDocument as WrappedDocument<OpenAttestationDocument>;
+      expect(utils.getDocumentData(v3WrappedDocument)).not.toHaveProperty("proof");
```
Comment (Nebulis) [inline src/shared/utils/__tests__/utils.test.ts:396]: Your don't test the rest is here

File: src/shared/utils/__tests__/utils.test.ts (hunk 6 above)
Comment (ChenRen93) [reply src/shared/utils/__tests__/utils.test.ts]: updated

File: src/shared/utils/__tests__/utils.test.ts (hunk 7 above)
Comment (ChenRen93) [reply src/shared/utils/__tests__/utils.test.ts]: added match inline. the object contains a default property that kept its original state before omit. I would like to ask if it is okay for the property to remain or we should remove that property before returning it back?

Comment (Nebulis) [reply src/shared/utils/__tests__/utils.test.ts]: nah it makes the test noisy. I just noticed how the json files are imported. Does `import bla from "x.json"` fix the issue? (no `import * as bla`)

Comment (ChenRen93) [reply src/shared/utils/__tests__/utils.test.ts]: okay, i update the create separate omit function that removes proof and default. changing to import bla from "x.json" doesn't fix the issue, the data is generalised as an object after importing, but the function only accepts WrappedDocument<OpenAttestationDocument>. i haven't found anyway to typecast during import

**Review by Nebulis — APPROVED** [review]

**Pull request merged**